src/Build/BackEnd/Components/Logging/ProjectLoggingContext.cs

```diff
@@ -96,7 +96,7 @@ internal ProjectLoggingContext(NodeLoggingContext nodeLoggingContext, BuildReque
 
             // If we are only logging critical events lets not pass back the items or properties
             if (!LoggingService.OnlyLogCriticalEvents &&
-                !LoggingService.IncludeEvaluationPropertiesAndItems &&
+                LoggingService.IncludeEvaluationPropertiesAndItems &&
```

Comment:
I am not at all sure about this change. Without it, the `Properties` of the `ProjectStartedEventArgs` are empty, and I can't manipulate the environment variables to tell the LoggingService to serialize all properties (or just a subset of properties) from within the logger, that's the wrong way around.

It feels like 

```csharp
        if (eventSource is IEventSource4 eventSource4)
        {
            eventSource4.IncludeEvaluationPropertiesAndItems();
        }
```

should be all we need to do to start getting this information, so I'm not entirely sure why I never do.

For lines 98-100, the conditions break down (before this change) while debugging as:

```csharp
if (!false &&
    !true && 
    (!false || false))
```

so the IncludeEvaluationPropertiesAndItems seems to be working against us here - perhaps because those are then intended to be on the ProjectEvaluationStart/Finish Events instead?

A bit further down, lines 118-121 evaluate as

```csharp
if (false &&
    !true && 
    false &&
    !false)
```

so in this case including the events is again causing what I expect to occur to not occur. And as mentioned, since I can't control `propertiesToSerialize` I can't ask for `GenerateFullPaths` to flow through :(

Reply:
The props and items are only on EvaluatinFinished event args.

@AR-May was recently fixing the env vars not being properly reflected in nodes - in this case (requested in logger initialization) it feels it should work, right?

Lastly - we'll be battling similar needs in Analyzers as well - where we might discover the need later during the evaluation, when it's already not possible to communicate the need to the node. So I'm wondering if it really has some measurable perf impact or if we should send the evaluated props and items in EventArgs allways unconditionally. @AR-May - would you want to create a workitem for yourself to try some perf runs with and without `MSBUILDLOGPROPERTIESANDITEMSAFTEREVALUATION` set to true/1?

Reply:
The story here is:

1. In the bad old days pre-binlog, the ProjectStarted event had a _text_ dump of items and properties.
2. Then we got the binlog and the viewer started _parsing_ that to recover stuff.
3. But that sucks! We have structured events!
4. So let's emit structured data on ProjectEvaluationFinished instead
5. Oh no, some _other_ loggers also parsed that text and broke.
6. So let's make the new behavior opt-in, and if any logger _hasn't_ opted in still emit that text.

So I told you wrong when I said it was on ProjectStarted, sorry!

Reply:
Haha, always happy to get more history.

So that means I need to figure out why my earlier attempt to get ProjectEvaluationFinishedEventArgs events never received any events, then.